### PR TITLE
Merge ign-gazebo6 ➡️  gz-sim7

### PR DESCRIPTION
**Pull request by azeey**
# ➡️  Forward port

Port `ign-gazebo6 ` ➡️  `gz-sim7`

Branch comparison: https://github.com/gazebosim/gz-sim/compare/gz-sim7...ign-gazebo6

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by iche033** [comment]
looks like the [fuel-tools8 debbuilds](https://build.osrfoundation.org/job/gz-fuel-tools8-debbuilder/) failed so the new API  needed by resource spawner is not available.

**Comment by iche033** [comment]
> looks like the [fuel-tools8 debbuilds](https://build.osrfoundation.org/job/gz-fuel-tools8-debbuilder/) failed so the new API needed by resource spawner is not available.

https://github.com/gazebo-release/gz-fuel-tools8-release/pull/14

**Review by iche033 — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

File: test/integration/CMakeLists.txt [hunk 1]
```diff
@@ -111,6 +112,7 @@ set(tests_needing_display
   shader_param_system.cc
   thermal_sensor_system.cc
   thermal_system.cc
+  triggered_camera.cc
```
Comment (azeey) [inline test/integration/CMakeLists.txt:115]: @iche033 This was removed from gz-sim7 for some reason. I added it back here, but if you know it should be removed, let me know.

Comment (iche033) [reply test/integration/CMakeLists.txt:115]: probably got lots in the forward ports. The test was forward ported [here](https://github.com/gazebosim/gz-sim/pull/1514/files#diff-612c1d43e6cd2b24070101fe026baf08ad423bbae381714d517a71d3c39495eb) but not the CMakeList.txt changes

**Comment by azeey** [comment]
[ServerRepeat/ServerFixture.SdfWithoutWorld/0](https://build.osrfoundation.org/job/ign_gazebo-pr-win/6073/testReport/junit/(root)/ServerRepeat_ServerFixture/SdfWithoutWorld_0/) is failing. Looking into it.

**Comment by azeey** [comment]
> [ServerRepeat/ServerFixture.SdfWithoutWorld/0](https://build.osrfoundation.org/job/ign_gazebo-pr-win/6073/testReport/junit/(root)/ServerRepeat_ServerFixture/SdfWithoutWorld_0/) is failing. Looking into it.

It turns out the test is not relevant in gz-sim7 because SDF files without `<world>` are supported in gz-sim7, so I've removed the test.

**Pull request merged**